src/Yarp.ReverseProxy.Swagger/ReverseProxyDocumentFilter.cs

```diff
@@ -82,6 +82,11 @@ public void Apply(OpenApiDocument swaggerDoc, DocumentFilterContext context)
                         var stream = httpClient.GetStreamAsync($"{destination.Value.Address}{swaggerPath}").Result;
                         var doc = new OpenApiStreamReader().Read(stream, out _);
 
+                        if (swagger.MetadataPath == swaggerPath)
```

Comment:
This code will be invoke each time in the cycle. Not need it. Better move it below after the cycle near by the "swaggerDoc.Paths = paths;"

Reply:
That's true, but I figured optimizing this would be premature and opted for simplicity :smiley:. 

To copy over the `Info` object into our swagger document, I need access to the OpenAPI spec object (`doc`).

```csharp
var stream = httpClient.GetStreamAsync($"{destination.Value.Address}{swaggerPath}").Result;
var doc = new OpenApiStreamReader().Read(stream, out _); // <-- parsed open api spec

if (swagger.MetadataPath == swaggerPath)
{
    swaggerDoc.Info = doc.Info; // <-- copy over doc info object
}
```

For each configured swagger path, we're making an HTTP request to get and parse the OpenAPI spec. We're going to be doing that for all swagger paths.

If I moved that code out of the loop, then I wouldn't have access to the `doc` object that I need. Obviously, we don't want another expensive HTTP call to get the OpenApi spec again. 

I think my best option would be to create a dictionary where the key is the swagger path and the value is the `doc` object. Therefore, the lookup is constant time. However, we incur more memory by adding the complexity of the dictionary.

I think most use cases will only have 1 swagger paths and at worst < 5. So I doubt we're optimizing by much. But, that's just my assumption 😄.

If you want me to use the dictionary instead, just let me know.

Reply:
Good answer! You're right.

I confused by "swaggerDoc.Info = doc.Info;" because filling of the swaggerDoc makes after the foreach cycle in one code block.

I suggest to define "info" variable above near by "var paths = new OpenApiPaths();", fill it here and use "swaggerDoc.Info = info;" near by "swaggerDoc.Paths = paths;".

This will make the code more friendly.

What do you think about it?

Reply:
Sure, that makes sense to me. That would match your existing code style better. 

I've updated the PR with the requested change.

Reply:
Cool! Thanks a lot! Today I'll release new version with your improvements.